### PR TITLE
ORC-1375: Cancel old running ci tasks when a pr has a new commit

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

This pr is aimed to cancel old running ci tasks when a pr has a new commit.

### Why are the changes needed?

The old tasks were based on outdated code which wasted resources and was pointless. Canceling tasks makes for a more efficient use of resources.

### How was this patch tested?

It doesn't seem to work immediately before merging, I verified this in other project.

### CODE REVIEW

**Comment by guiyanakuang** [comment]
This pr modifies `build_and_test.yml`, but it doesn't seem to take effect immediately.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -14,6 +14,11 @@ on:
     branches:
     - main
 
+# Cancel previous PR build and test
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.event.number || github.sha }}
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:19]: Is it enough to isolate `Pull Request`, @guiyanakuang ?

Comment (guiyanakuang) [reply .github/workflows/build_and_test.yml:19]: I think it's enough, almost all contributions are currently committed by pull request. Also when we use to receive other events, we can degrade to grouping using `github.sha`.

**Review by dongjoon-hyun — APPROVED** [review]
Got it. Let's try it in `main` branch. Thank you, @guiyanakuang .

**Comment by dongjoon-hyun** [comment]
Also, cc @williamhyun , too.

**Comment by guiyanakuang** [comment]
image
Looks like it's working well. Thanks for the review and merge, @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Thank you for checking. :)

**Pull request closed without merging**